src/main/java/util/RefreshTimer.java

```diff
@@ -0,0 +1,105 @@
+package util;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * This class is a subclass of TickingTimer. It contains logic to facilitate dynamic period of the timer.
+ */
+public class RefreshTimer extends TickingTimer{
+
+    /**
+     * The buffer time to be added to ensure refresh happens after the api quota renewal.
+     */
+    public static final int BUFFER_TIME = 1;
```

Comment:
I was thinking that the buffer-time should be configurable. If the programmer wish to give the api server some more time before the next pull. 
Even we don't want to allow it to be configurable, having it as a constant will help the code to be more understandable?